model/user.go

```diff
@@ -47,6 +47,77 @@ const (
 	USER_PASSWORD_MAX_LENGTH  = 72
 )
 
+// This is a struct for a user interaction
+// A user interaction will represent one row in the results of the
+// GetUserInteractions query in store/sqlstore/user_store.go
+type UserInteraction struct {
+	// The type of interaction
+	// enum('Reaction', 'Mention', 'Reply', 'DirectMessage')
+	InteractionType string `json:"interaction_type"`
+
+	// The username of the user that this interaction took place with
+	Username string `json:"username"`
+
+	// Denotes whether or not the current user was the recipient of this interaction
+	IsRecipient bool `json:"is_recipient"`
+
+	// The context in which this interaction occurred (either 'Course' or the learning group type's prefix (ex. 'plg'))
+	Context string `json:"context"`
+
+	// The type of channel that this interaction took place in
+	// enum('P', 'O', 'D', 'G')
+	//     P = private channel
+	//     O = public channel
+	//     D = direct message (1 on 1)
+	//     G = group direct message
+	ChannelType string `json:"channel_type"`
+
+	// The slug name of the channel that this interaction took place in (Channels.Name)
+	ChannelName string `json:"channel_name"`
+}
+
+// This is a struct for a learning group type
+// Each learning group type is derived from personal channel names found in the config
+type LearningGroupType struct {
+	// ex. 'capstone'
+	Prefix string `json:"prefix"`
+}
+
+// This is a struct for all user analytics data sets
+// This is the struct that gets returned to mattermost-redux in the users/{userId}/teams/{teamId}/useranalytics request
+// **Note: Using pointers as property types allows for null values
+type UserAnalytics struct {
+	TargetUser         string               `json:"TargetUser"`
+	UserInteractions   []*UserInteraction   `json:"UserInteractions"`
+	UserLearningGroups []*LearningGroup     `json:"UserLearningGroups"`
+	LearningGroupTypes []*LearningGroupType `json:"LearningGroupTypes"`
+	// Add more datasets here
+}
+
+// This is a struct for a learning group (ex. capstone)
+// **Note: Using pointers as property types allows for null values
+type LearningGroup struct {
+	// The name of the learning group (the channel name minus the learning group type's prefix)
+	//     ex. plg-30 => 30
+	LearningGroupName string `json:"learning_group_name"`
+
+	// The type of learning group (personal channel name from the config)
+	//     ex. plg
+	LearningGroupType string `json:"learning_group_type"`
```

Comment:
This maybe should be a LearningGroupType, but right now it is just the LearningGroupType's prefix. I sort of feel that either the name or the type should be changed, because otherwise it's confusing. I think just changing the name is reasonable, back to LearningGroupPrefix which can be used to find the LearningGroupType in the array of them that is returned if that was desired (say to get other properties, which might exist in the future)